### DISCUSSION TITLE
More than 6 CT's.  Stacking HAT's.  Multiple PI Monitors ???

### DISCUSSION THREAD

**Discussion by wotseurba** [David00/rpi-power-monitor#127] [Q&A]
I have the need to employ more than the 6 CT's allowed with the Monitor Board.

As noted from a previous discussion that stacking the Monitor HAT's can't be done without revision to the design.  Is the prohibition purely due to components on the board not allowing connection to the GPIO header?  OR, is it due to code or electronics of the monitor board?

A second option which was discussed is using multiple PI Monitors along with a remote InfluxDB Server.  My question is:  Can I use my presently operating PI Monitor as the remote InfluxDB server, and having the 'secondary' PI monitors send their data to it?

Thank you in advance.  I'm neither a software, or electrical engineer, but rather an avid DIY enthusiast who enjoys a challenging project from which to learn.  I'm thrilled with my running Energy Monitor, and applaud David for the superb work to make this project, and share his work with us.

**Comment by David00** [comment]
Hi @wotseurba, great question, and thanks for opening a discussion on this.


> As noted from a previous discussion that stacking the Monitor HAT's can't be done without revision to the design. Is the prohibition purely due to components on the board not allowing connection to the GPIO header? OR, is it due to code or electronics of the monitor board?

The Pi itself would support using two boards with its two SPI chip select lines. This would theoretically allow the Pi to switch between one board or another (but not both at the same time).  Therefore, the limitation is primarily due to the power monitor PCB being hardwired to use a single SPI chip select line.  A future hardware revision could get around this by exposing a jumper or microswitch on the PCB, allowing the user to configure the board for CS0 or CS1, and then the boards could be stacked (with a passthrough header installed on the PCB as well)

> A second option which was discussed is using multiple PI Monitors along with a remote InfluxDB Server. My question is: Can I use my presently operating PI Monitor as the remote InfluxDB server, and having the 'secondary' PI monitors send their data to it?

Yes, you can do this, but definitely make sure you have a "high endurance" microSD card (or better yet - a USB to NVME SSD) before doing so. If you use a standard microSD card, it **will** fail in under 1 year.  The high endurance cards will also fail eventually, highlighting the need for [automatic backups](https://david00.github.io/rpi-power-monitor/docs/v0.3.0/configuration#enabling-automatic-backups), but they seem to last a **lot** longer. My high endurance card has been going strong for I think at least 2 years now.

I'd also like to present a third option - 
I've been slowly but surely working on a 14 channel board, with 4x the resolution and more than 2x the sampling speed.  This is essentially a new project with a new PCB, completely redesigned and rewritten software, a few more advanced features, along with a webUI to setup the and monitor power monitor.  The current sensors are the same, though, so it is just a PCB swap on your Pi and a new microSD card flash to setup. The hardware is ready to go and I am working on maturing the software side of things as well, but the basic power monitoring functionality has been done for a while.  It is in a closed-beta stage right now with several folks from around the world using it and providing feedback.  If this is something you'd be interested in getting your hands on, please email me at github (at) dalbrecht.tech and I'll share more information.

**Reply by wotseurba** [reply]
Thank you for the detailed explanation and recommendations.  My current monitor is running with your DIY board, on a Pi 3B, which I had laying around.  My plan is to build up a Pi4 with a NVME SSD, as the boot drive, where the InfluxDB will reside.  I will experiment with PI Zero 2W's as the "remote" monitors, and have them write their data collection to the "mother" Pi4.  

As a side note, I use Tailscale as a VPN to access my Raspberry Pi's outside my network and have been completely satisfied.

**Reply by David00** [reply]
No problem!  That sounds great - the new 14 channel version of the project works very well on the Pi Zero 2W and its limited amount of RAM.  It's been awhile since I have tested the 6-channel version on a Pi Zero 2W, but if you send the measurements from the Zero 2W to the remote Pi 4, you can disable InfluxDB and Grafana on the Zero, giving you the best chance of success.

Thanks for letting me know about Tailscale. It looks interesting.

**Reply by cbrodine** [reply]
I hope this new project is going well! I'm very interested in monitoring WAY more than 14 channel board would allow. I own a rural property with 3 panels in different buildings, so I'm thinking of an ultimate goal of multiple installations on each panel to monitor all my 2 pole breakers and a few of my single pole breakers. (I've been using Shelly energy monitors behind a few outlets for some individual loads already)

I'm ultimately trying to create my own demand management system
